### ISSUE TITLE
Upgrade kubernetes libraries and get rid of `replace` in go.mod

### ISSUE THREAD

**Issue by jimmykarily** [SUSE/carrier#93]
When installing in some kubernetes versions we get this error after install:

```
error initializing cli: can't get gitea url: failed to list ingresses for gitea: the server could not find the requested resource
```
there is a hypothesis that this is due to our kubernetes libraries not recognizing the CRDs in newer versions of kubernetes. To verify we need to upgrade our libraries in go.mod

Right now we have `replace` statements in the go.mod file. Removing them and running `go mod tidy` reveals the problem.

```
~/workspace/suse/carrier/cli (main)*$ go mod tidy
github.com/suse/carrier/cli/kubernetes imports
	k8s.io/client-go/kubernetes/scheme imports
	k8s.io/api/auditregistration/v1alpha1: module k8s.io/api@latest found (v0.20.2), but does not contain package k8s.io/api/auditregistration/v1alpha1
github.com/suse/carrier/cli/kubernetes/config imports
	k8s.io/client-go/discovery imports
	github.com/googleapis/gnostic/OpenAPIv2: module github.com/googleapis/gnostic@latest found (v0.5.3), but does not contain package github.com/googleapis/gnostic/OpenAPIv2
```
which leads to this issue: https://github.com/kubernetes/client-go/issues/741

Again, upgrading to newer versions would probably solve this issue but it's probably due to `quarks-utils` needing a very old version of `client-go` that we can't.

We remove our dependency to `quarks-utils` [here](https://github.com/SUSE/carrier/pull/91/files#diff-37a093b0dd13152e13290bd482db8bc68232329d6f3e9053f04c4e02eb910959L15) but we still get it indirectly because we import `eirini` which imports `eirinix` which imports `quarks-utils`.

The reason we import Eirini is this line here: https://github.com/SUSE/carrier/blob/main/cli/paas/client.go#L177

This is hardly a good reason for all this mess so maybe we can find another way to delete LRPs without the need to import the code generated client from eirini. (How is `kubectl` able to delete any Custom Resource without importing anything? Maybe  [apiextensions-apiserver project described here](https://www.velotio.com/engineering-blog/extending-kubernetes-apis-with-custom-resource-definitions-crds) is useful?)

Also, [this](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/config#GetConfig) seems to be a better way to setup a kubernetes client [than the one we use](https://github.com/SUSE/carrier/blob/main/cli/cmd/commons.go#L28)

**Comment by andreas-kupries** [comment]
Note that the referenced `CreateKubeClient` function is not used in our code (anymore). I believe its use went away with going to Vlad's `cli/kubernetes/config` package.

**Comment by jimmykarily** [comment]
Maybe we can use a dynamic client? https://ymmt2005.hatenablog.com/entry/2020/04/14/An_example_of_using_dynamic_client_of_k8s.io/client-go

**Comment by andreas-kupries** [comment]
:eyes: :link: ... Back ... Agreed, that looks interesting and promising.

**Comment by jimmykarily** [comment]
Removed the eirini dependency here: https://github.com/SUSE/carrier/tree/bump_kubernetes_libraries
Will continue on the same branch to try to upgrade the kubernetes libraries

**Comment by jimmykarily** [comment]
Tried on gke:  europe-west3-c  1.18.14-gke.1200  35.242.244.7  n1-highcpu-8  1.18.14-gke.1200  1        

and still getting this error:

```
✔️  Carrier installed.
System domain: 34.89.246.12.nip.io
error initializing cli: can't get gitea url: failed to list ingresses for gitea: the server could not find the requested resource
```

**Comment by jimmykarily** [comment]
It seems that we have 2 types of ingress resources known to the cluster: 
```
~/workspace/suse/carrier/cli (bump_kubernetes_libraries)$ kubectl  api-resources | grep ingresses
ingresses                         ing          extensions                     true         Ingress
ingresses                         ing          networking.k8s.io              true         Ingress
```

and kubectl makes no distinction between them:

```
~/workspace/suse/carrier/cli (bump_kubernetes_libraries)$ kubectl  get ingresses.networking.k8s.io --all-namespaces
NAMESPACE          NAME               CLASS    HOSTS                        ADDRESS   PORTS   AGE
gitea              gitea              <none>   gitea.34.89.246.12.nip.io              80      90m
tekton-pipelines   tekton-dashboard   <none>   tekton.34.89.246.12.nip.io             80      87m
~/workspace/suse/carrier/cli (bump_kubernetes_libraries)$ kubectl  get ingresses.extensions --all-namespaces
NAMESPACE          NAME               CLASS    HOSTS                        ADDRESS   PORTS   AGE
gitea              gitea              <none>   gitea.34.89.246.12.nip.io              80      90m
tekton-pipelines   tekton-dashboard   <none>   tekton.34.89.246.12.nip.io             80      87m
```

**Comment by andreas-kupries** [comment]
:facepalm:

**Comment by jimmykarily** [comment]
ok I created a k3d cluster on version 1.18.15 (same as on gke) and I get the same error. So if we use the Networking v1 library in our code to create/access Ingresses, it's not going to work on clusters with version < 1.19 . It does work on newer clusters. We have the following  2 options:

- Either access Ingresses using v1beta1 (like we did so far with Tekton)
- Have a conditional in the code and use Networking v1 (which is the recommended way in newer versions) when cluster version is >= 1.19 and fallback to v1beta1 on older clusters.
